File: PathoFroh/src/main/java/com/patho/main/util/helper/UniqueRequestID.java
package com.patho.main.util.helper;

import java.math.BigInteger;
import java.security.SecureRandom;
import java.util.Map;

import javax.faces.context.FacesContext;

import com.patho.main.model.patient.Diagnosis;
import com.patho.main.util.exception.CustomNotUniqueReqest;

import lombok.Getter;
import lombok.Setter;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class UniqueRequestID {

	private final static String SUBMITTEDID = "submittedRequestID";

	private SecureRandom random = new SecureRandom();

	@Getter
	@Setter
	private String uniqueRequestID;

	@Getter
	@Setter
	private String submittedRequestID;

	@Getter
	@Setter
	private boolean enabled;

	public UniqueRequestID() {
		setUniqueRequestID("");
	}

	public void nextUniqueRequestID() {
		setUniqueRequestID(new BigInteger(130, random).toString(32));
		log.debug("New Unique ID generated");
	}

	public void checkUniqueRequestID(boolean toClose) {
		// disables uniqueCheck
		checkUniqueRequestID(toClose, false);
	}

	public void checkUniqueRequestID(boolean toClose, boolean enabled) {

		FacesContext fc = FacesContext.getCurrentInstance();
		Map<String, String> params = fc.getExternalContext().getRequestParameterMap();
		String submittedRequestID = params.get(SUBMITTEDID);

		if (submittedRequestID == null || submittedRequestID.isEmpty()) {
			log.debug("No ID submitted");
			throw new CustomNotUniqueReqest(toClose);
		}

		if (uniqueRequestID.isEmpty() || !uniqueRequestID.equals(submittedRequestID)) {
			log.debug("ID does not equal");
			throw new CustomNotUniqueReqest(toClose);
		}

		log.debug("Unique ID matched");

		setUniqueRequestID("");
		setEnabled(enabled);
	}
}

File: PathoFroh/src/main/kotlin/com/patho/main/util/dialog/UniqueRequestID.kt
package com.patho.main.util.dialog

import com.patho.main.util.exception.CustomNotUniqueReqest
import org.slf4j.LoggerFactory
import java.math.BigInteger
import java.security.SecureRandom
import javax.faces.context.FacesContext

class UniqueRequestID {

    protected val logger = LoggerFactory.getLogger(this.javaClass)

    private val SUBMITTEDID = "submittedRequestID"

    private val random = SecureRandom()


    var uniqueRequestID: String = ""

    var enabled: Boolean = false

    /**
     * Generates the next unique id
     */
    fun nextUniqueRequestID() {
        uniqueRequestID = BigInteger(130, random).toString(32)
        logger.debug("New Unique ID generated -> $uniqueRequestID")
    }

    /**
     * Checks if the request id matches
     */
    fun checkUniqueRequestID(toClose: Boolean) {
        // disables uniqueCheck
        checkUniqueRequestID(toClose, false)
    }

    /**
     * Checks if the request id matches
     */
    fun checkUniqueRequestID(toClose: Boolean, enabled: Boolean) {

        val fc = FacesContext.getCurrentInstance()
        val params = fc.externalContext.requestParameterMap
        val submittedRequestID = params[SUBMITTEDID]

        if (submittedRequestID == null || submittedRequestID.isEmpty()) {
            logger.debug("No ID submitted")
            throw CustomNotUniqueReqest(toClose)
        }

        if (uniqueRequestID.isEmpty() || uniqueRequestID != submittedRequestID) {
            logger.debug("ID does not equal")
            throw CustomNotUniqueReqest(toClose)
        }

        logger.debug("Unique ID matched")

        uniqueRequestID = ""
        this.enabled = enabled
    }
}